### PR TITLE
refactor: GraphGenerator with accurate BirthRounds

### LINKED ISSUE #11826: TestFixture: Expand GraphGenerator with ability to update birthrounds using internal consensus

### PR DESCRIPTION
**Pull request by edward-swirldslabs**
Fixes #11826

### CODE REVIEW

**Comment by cody-littley** [comment]
Friendly reminder to mark yourself as the assignee. 😉

File: platform-sdk/swirlds-platform-core/src/testFixtures/java/com/swirlds/platform/test/fixtures/event/source/EventSource.java [hunk 1]
```diff
@@ -106,18 +106,19 @@ default T setNewEventWeight(final double newEventWeight) {
...
+     * @param birthRound  the pending consensus round when the event was created
      * @return The random event that was generated.
      */
     IndexedEvent generateEvent(
-            final Random random, final long eventIndex, final EventSource<?> otherParent, final Instant timestamp);
+            final Random random,
+            final long eventIndex,
+            final EventSource<?> otherParent,
+            final Instant timestamp,
+            final long birthRound);
```
[the remark this answers is not in the document]
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/testFixtures/java/com/swirlds/platform/test/fixtures/event/source/EventSource.java]: added

**Review by cody-littley — DISMISSED** [review]

File: platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/sync/SyncTests.java [hunk 2]
```diff
@@ -721,36 +721,33 @@ private void runFallenBehindTest(
...
     @MethodSource("bothAncientModes")
     void testBarelyNotFallenBehind(@NonNull final AncientMode ancientMode) throws Exception {
-        final SyncTestParams params = new SyncTestParams(4, 200, 200, 0, ancientMode);
+        final SyncTestParams params = new SyncTestParams(4, 2000, 2000, 0, ancientMode);
         final SyncTestExecutor executor = new SyncTestExecutor(params);
 
         executor.setEventWindowDefinitions((caller, listener) -> {
             long listenerMaxIndicator =
                     SyncTestUtils.getMaxIndicator(listener.getShadowGraph().getTips(), ancientMode);
+            System.out.println(ancientMode + ": listenerMaxIndicator : " + listenerMaxIndicator);
```
Comment (alittley) [inline platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/sync/SyncTests.java:730]: Are these prints intentionally left here?

**Review by alittley — APPROVED** [review]

File: platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/sync/SyncTests.java (hunk 2 above)
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/sync/SyncTests.java:730]: nope, deleting them

**Review by alittley — APPROVED** [review]

**Review by cody-littley — APPROVED** [review]

**Pull request merged**